app/models/status.rb

```diff
@@ -92,6 +93,8 @@ class Status < ApplicationRecord
   scope :including_silenced_accounts, -> { left_outer_joins(:account).where.not(accounts: { silenced_at: nil }) }
   scope :not_excluded_by_account, ->(account) { where.not(account_id: account.excluded_from_timeline_account_ids) }
   scope :not_domain_blocked_by_account, ->(account) { account.excluded_from_timeline_domains.blank? ? left_outer_joins(:account) : left_outer_joins(:account).where('accounts.domain IS NULL OR accounts.domain NOT IN (?)', account.excluded_from_timeline_domains) }
+  scope :hidden, ->(account) { joins(:hidden_statuses).where('hidden_statuses.account_id': account.id) }
+  scope :not_hidden, ->(account) { where.not(id: hidden(account)) }
```

Comment:
I have a hard time thinking about the performance implications of that query, tbh.